Synview-frontend/src/slices/syncGithubModalSlice.ts

```diff
@@ -0,0 +1,33 @@
+import { createSlice } from "@reduxjs/toolkit";
+import type { PayloadAction } from "@reduxjs/toolkit";
+
+interface githubModalState {
+  project_id: number | undefined;
+  user_id: number | undefined;
```

Comment:
nit: An interface like this is kinda a type, it should always be capitalized - `GithubModalState`

Also - would it ever be correct to have a `GithubModalState` without a project ID or a user ID? These should basically be required right? If so - I would remove the `undefined` here an enforce non-nullability